shiny/ui/_theme.py

```diff
@@ -0,0 +1,506 @@
+from __future__ import annotations
+
+import copy
+import os
+import pathlib
+import tempfile
+from textwrap import dedent
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    Literal,
+    Optional,
+    Sequence,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from htmltools import HTMLDependency
+from packaging.version import Version
+
+from .._docstring import no_example
+from .._typing_extensions import NotRequired, TypedDict
+from .._versions import bootstrap
+from ._theme_presets import (
+    ShinyThemePreset,
+    shiny_theme_presets,
+    shiny_theme_presets_bundled,
+)
+from ._utils import path_pkg_www
+
+T = TypeVar("T", bound="Theme")
```

Comment:
Seems like you could get rid of this?

Reply:
Required for method chaining of the `.add_*()` methods